File: src/de/sciencecomputing/venuside/ContextSyntaxHighlighterFactory.java
package de.sciencecomputing.venuside;

import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;

import org.jetbrains.annotations.NotNull;

public class ContextSyntaxHighlighterFactory extends SyntaxHighlighterFactory {
    @NotNull
    @Override
    public SyntaxHighlighter getSyntaxHighlighter(Project project, VirtualFile file) {
        return new ContextSyntaxHighlighter();
    }
}

File: src/de/sciencecomputing/venuside/ContextSyntaxHighlighterFactory.kt
package de.sciencecomputing.venuside

import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

class ContextSyntaxHighlighterFactory : SyntaxHighlighterFactory() {
    override fun getSyntaxHighlighter(project: Project?, file: VirtualFile?)
            : SyntaxHighlighter = ContextSyntaxHighlighter()
}